src/test/java/com/zipcodewilmington/scientific_calculator/TestMainApplication.java

```diff
@@ -1,7 +1,323 @@
+        assertEquals(Double.valueOf(2.0), myCalc.divide(8,4), 0.00001);
+        assertEquals(Double.valueOf(0.25), myCalc.divide(1,4), 0.00001);
+    }
+
+    @org.junit.Test
+    public void testExponential() {
+        Calculator myCalc = new Calculator();
+        assertEquals(Double.valueOf(9.0), myCalc.exponential(3,2), 0.00001);
+        assertEquals(Double.valueOf(8.0), myCalc.exponential(2,3), 0.00001);
+        assertEquals(Double.valueOf(1.0), myCalc.exponential(2,0), 0.00001);
+    }
+
+    @org.junit.Test
+    public void testSquare() {
+        Calculator myCalc = new Calculator();
+        assertEquals(Double.valueOf(4.0), myCalc.square(2), 0.00001);
+        assertEquals(Double.valueOf(1.0), myCalc.square(1), 0.00001);
+        assertEquals(Double.valueOf(1600.0), myCalc.square(40), 0.00001);
+    }
+
+    @org.junit.Test
+    public void testSquareRoot() {
+        Calculator myCalc = new Calculator();
+        assertEquals(Double.valueOf(1.41421), myCalc.squareRoot(2), 0.00001);
+        assertEquals(Double.valueOf(2.0), myCalc.squareRoot(4), 0.00001);
+        assertEquals(Double.valueOf(4.0), myCalc.squareRoot(16), 0.00001);
+    }
+
+    @org.junit.Test
+    public void testInverse() {
+        Calculator myCalc = new Calculator();
+        assertEquals(Double.valueOf(0.5), myCalc.inverse(2), 0.00001);
+        assertEquals(Double.valueOf(2.0), myCalc.inverse(0.5), 0.00001);
+        assertEquals(Double.valueOf(0.1), myCalc.inverse(10), 0.00001);
+    }
+
+    @org.junit.Test
+    public void testInvert() {
+        Calculator myCalc = new Calculator();
+        assertEquals(Double.valueOf(-2.0), myCalc.invert(2), 0.00001);
+        assertEquals(Double.valueOf(3.0), myCalc.invert(-3), 0.00001);
+        assertEquals(Double.valueOf(0.0), myCalc.invert(0), 0.00001);
+    }
+
+    @org.junit.Test
+    public void testSine() {
+        Calculator myCalc = new Calculator();
+        assertEquals(Double.valueOf(0), myCalc.sine(0), 0.00001);
+        assertEquals(Double.valueOf(0.84147), myCalc.sine(1), 0.00001);
+        assertEquals(Double.valueOf(0.9093), myCalc.sine(2), 0.00001);
+    }
+
+    @org.junit.Test
+    public void testCosine() {
+        Calculator myCalc = new Calculator();
+        assertEquals(Double.valueOf(1.0), myCalc.cosine(0), 0.00001);
+        assertEquals(Double.valueOf(0.5403), myCalc.cosine(1), 0.00001);
+        assertEquals(Double.valueOf(-0.41615), myCalc.cosine(2), 0.00001);
+    }
+
+    @org.junit.Test
+    public void testTangent() {
+        Calculator myCalc = new Calculator();
+        assertEquals(Double.valueOf(0.0), myCalc.tangent(0), 0.00001);
+        assertEquals(Double.valueOf(1.55741), myCalc.tangent(1), 0.00001);
+        assertEquals(Double.valueOf(-2.18504), myCalc.tangent(2), 0.00001);
+    }
+
+    @org.junit.Test
+    public void testInverseSine() {
+        Calculator myCalc = new Calculator();
+        assertEquals(Double.valueOf(-1.5708), myCalc.inverseSine(-1), 0.00001);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [CNT: Rough value of known constant found (CNT_ROUGH_CONSTANT_VALUE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#cnt-rough-value-of-known-constant-found-cnt-rough-constant-value).
It's recommended to use the predefined library constant for code clarity and better precision.